lib/phoenix/token.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Phoenix.Token do
+  @moduledoc """
+  Will generate, verify, and decrypt tokens for use in channel or API 
+  authentication. Typically you will want to store the basic information
+  you need to authorize a user. Typically as simple as a user id in a 
+  database. 
+
+      iex> user_id = 1
+      iex> token = encrypt(user_id, endpoint)
+      iex> user_id == decrypt(token, endpoint)
+      true
```

Comment:
Let's get some high-level examples of usage in the moduledoc. Maybe even showing using in a `def join(...`. We also need to add examples to the `gen_token` and `verify_token` with their return values.